bcipy/signal_model/mach_learning/classifier/function_classifier.py

```diff
@@ -185,3 +186,114 @@ def fit_transform(self, x, y, p=[]):
 
         return self.transform(x)
 
+
+class MDiscriminantAnalysis:
+    """ Robust version of RDA where every estimated statistic is estimated
+    Ref:
+        Friedman, Jerome H. "Regularized discriminant analysis."
+        Journal of the American statistical association 84.405 (1989): 165-175
+
+        and
+
+        Kadioglu, Berkan "M Estimation Based Robust Subspace learning for Brain Computer Interfaces"
+
+    Attr:
+        lam(float): shrinkage param
+        gam(float): threshold param (a.k.a. regularization param)
+        means(list): Means' of each channel
+        covariances(list): Covariances' of each channel
+        S_matrices(list): Covariance times N for each class
+        S(list): Covariance times N for all data
+        N_list(list): List of number of samples of each class
+        class_list(list): List of classes
+        priors(list): List of priors for each class
+        inv_reg_covariances(list): List of inverse regularized covariances for each class
+
+    """
+    def __init__(self):
+        # means and covariances of each channel with the order inherent in data.
+        self.means = []
+        self.covariances = []
+        self.S_matrices = []  # Covariance times N for each class
```

Comment:
lower case and chose a more descriptive name

Reply:
This is the definitoin of Umut Orhan in his thesis, he defined these matrices as S. I will change it. I don't feel comfortable changing it.

Reply:
There is a difference between mathematical notation and useful variable naming in code. These are rules we agreed to in the beginning of this project. If you'd like a larger group discussion around this please add to the agenda and consult your team lead and advisor.